src/itensor.jl

```diff
@@ -955,12 +955,18 @@ Base.:/(A::ITensor, x::Number) = A*(1.0/x)
 Base.:-(A::ITensor) = itensor(-tensor(A))
 
 function Base.:+(A::ITensor, B::ITensor)
+  if order(A) != order(B)
```

Comment:
We could instead make this a compile time check by defining this function as:
```julia
Base.:+(A::ITensor{N}, B::ITensor{N}) where {N}
```
then if we want we can add our own error message by defining the more general version:
```julia
Base.:+(A::ITensor, B::ITensor) = error("different number of indices when adding ITensors")
```
which it should call as a fallback.